packages/xo-server/src/xo-mixins/metadata-backups.js

```diff
@@ -328,24 +328,20 @@ export default class metadataBackup {
           return
         }
 
-        await Promise.all(
-          files.map(async ({ executeBackup, dir, retention }) => {
-            await executeBackup(handler)
-
-            // deleting old backups
-            await handler.list(dir).then(list => {
-              list.sort()
-              list = list
-                .filter(timestampDir => timestampReg.test(timestampDir))
-                .slice(0, -retention)
-              return Promise.all(
-                list.map(timestampDir =>
-                  handler.rmtree(`${dir}/${timestampDir}`)
-                )
-              )
-            })
+        await asyncMap(files, async ({ executeBackup, dir, retention }) => {
+          await executeBackup(handler)
+
+          // deleting old backups
+          await handler.list(dir).then(list => {
+            list.sort()
+            list = list
+              .filter(timestampDir => timestampReg.test(timestampDir))
+              .slice(0, -retention)
+            return Promise.all(
```

Comment:
Still `Promise.all`?

Reply:
If a file deletion fails, should it impact the backup status?